src/braket/circuits/gates.py

```diff
@@ -215,6 +215,123 @@ def i(
+            .. math:: \mathtt{gphase}(\gamma) = e^(i \gamma) I_m.
+
+        Returns:
+            np.ndarray: GPhase Matrix
+        """
+        return np.exp(1j * self.angle) * np.eye(1, dtype=complex)
+
+    def bind_values(self, **kwargs) -> AngledGate:
+        return get_angle(self, **kwargs)
+
+    @staticmethod
+    def fixed_qubit_count() -> int:
+        return 0
+
+    @staticmethod
+    @circuit.subroutine(register=True)
+    def gphase(
+        angle: Union[FreeParameterExpression, float],
+        *,
+        control: Optional[QubitSetInput] = None,
+        control_state: Optional[BasisStateInput] = None,
+        power: float = 1,
+    ) -> Instruction | Iterable[Instruction]:
+        """Registers this function into the circuit class.
+
+        Args:
+            angle (Union[FreeParameterExpression, float]): Phase in radians.
+            control (Optional[QubitSetInput]): Control qubit(s). Default None.
+            control_state (Optional[BasisStateInput]): Quantum state on which to control the
+                operation. Must be a binary sequence of same length as number of qubits in
+                `control`. Will be ignored if `control` is not present. May be represented as a
+                string, list, or int. For example "0101", [0, 1, 0, 1], 5 all represent
+                controlling on qubits 0 and 2 being in the \\|0⟩ state and qubits 1 and 3 being
+                in the \\|1⟩ state. Default "1" * len(control).
+            power (float): Integer or fractional power to raise the gate to. Negative
+                powers will be split into an inverse, accompanied by the positive power.
+                Default 1.
+
+        Returns:
+            Instruction | Iterable[Instruction]: GPhase instruction.
+
+        Examples:
+            >>> circ = Circuit().gphase(0.45)
+        """
+        if control is not None:
+            control_qubits = QubitSet(control)
+
+            if control_state is None:
+                control_state = len(control_qubits)
+            control_basis_state = list(BasisState(control_state, len(control_qubits)).as_tuple)
+
+            if not any(control_basis_state):
+                return [
+                    X.x(control_qubits[-1]),
+                    PhaseShift.phaseshift(
+                        control_qubits[-1],
+                        angle,
+                        control=control_qubits[:-1],
+                        control_state=control_basis_state[:-1],
+                        power=power,
+                    ),
+                    X.x(control_qubits[-1]),
+                ]
+
+            highest_control_qubit = max(
+                [qubit for qubit, state in zip(control_qubits, control_basis_state) if state]
+            )
+            highest_control_qubit_index = control_qubits.index(highest_control_qubit)
+
+            control_qubits.pop(highest_control_qubit_index)
+            control_basis_state.pop(highest_control_qubit_index)
```

Comment:
I'm a bit confused by this logic. The ordering of the control modifiers is well-defined, with the rightmost qubit/modifier getting applied first. This shouldn't change depending on whether that modifier is ctrl or negctrl.

Reply:
An operation with `control=[2, 5, 3], control_state="011"` should first apply positive control to qubit 3, then positive control to qubit 5, then negative control to qubit 2. It looks like this code uses qubit value (rather than position in the control list) to sort, starting in this example with control on qubit 5. The order of ctr/negctrl applied should be 3, 5, 2, regardless of control state.

I think we want something like
```
if control_qubits:
    phase_target = control_qubits[-1]
    target_control_state = control_basis_state[-1]
    phase_instruction = PhaseShift.phaseshift(
        phase_target,
        angle,
        control=control_qubits[:-1],
        control_state=control_basis_state[:-1],
        power=power,
    ),
    cphase_instructions = (
        [phase_instruction]
        if target_control_state:
        else [X.x(phase_target), phase_instruction, X.x(phase_target)]
    )
```
Though maybe we want to add another gate for "neg" phaseshift instead of relying on the Xs

Reply:
I don't believe the added code complexity is worth it here. It may also confuse customers to see their qubits misordered, even if the operations are commutative. If the x-phase-x is dispreferred, then I believe we should use a unitary or add a gate.

Maybe @speller26 can weigh in here too